gbrl/common/compression.py

```
@@ -0,0 +1,612 @@
+    and 'best_k' (learn which k trees to remove) compression methods.
+    
+    Attributes:
+        compression (CompressionMethod): The compression method implementation (FirstK or BestK).
+        optimizer (th.optim.Optimizer): Optimizer for learnable parameters.
+        device (str): Device for computation ('cpu' or 'cuda').
+        method (str): Compression method name ('first_k' or 'best_k').
+        use_W (bool): Whether to use learnable correction matrix W.
+        gradient_steps (int): Number of optimization steps for compression.
+    """
+    
+    def __init__(self, k: int, gradient_steps: int, n_trees: int, n_leaves_per_tree: Union[np.ndarray, th.Tensor],
```

Ruff (0.14.14)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)